### PR TITLE
Tests - Boosted code coverage back up to > 80%.

### PR DESCRIPTION
**Pull request by thehenrytsai**

### CODE REVIEW

File: tests/core/Blockchain.spec.ts [hunk 1]
```diff
@@ -0,0 +1,227 @@
+import Blockchain from '../../lib/core/Blockchain';
+import ReadableStream from '../../lib/common/ReadableStream';
+import SharedErrorCode from '../../lib/common/SharedErrorCode';
+import TransactionModel from '../../lib/common/models/TransactionModel';
+// import { FetchResultCode } from '../../lib/common/FetchResultCode';
```
Comment (OR13) [inline tests/core/Blockchain.spec.ts]: consider removing unused.

File: tests/core/Blockchain.spec.ts [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+      };
+      spyOn(blockchainClient as any, 'fetch').and.returnValue(Promise.resolve(mockFetchResponse));
+      spyOn(ReadableStream, 'readAll').and.returnValue(Promise.resolve(JSON.stringify({
+        code: 'unused'
+      })));
+
+      try {
+        await blockchainClient.read(1, 'Unused transaction hash.');
+      } catch {
+        // Throwing error is the expected case.
```
Comment (OR13) [inline tests/core/Blockchain.spec.ts]: option to assert more about the nature of the error.

File: tests/core/Blockchain.spec.ts [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+
+      fail();
+    });
+
+    it('should throw if not both the transaction time and hash are given.', async () => {
+      const blockchainClient = new Blockchain('Unused URI');
+      try {
+        await blockchainClient.read(1, undefined);
+      } catch {
+        // Throwing error is the expected case.
```
Comment (OR13) [inline tests/core/Blockchain.spec.ts]: option to assert more about the nature of the error.

**Comment by OR13** [comment]
This kind of PR makes me super happy :) 🚀

File: tests/core/Blockchain.spec.ts (hunk 1 above)
Comment (thehenrytsai) [reply tests/core/Blockchain.spec.ts]: Mistake, will remove.

File: tests/core/Blockchain.spec.ts (hunk 2 above)
Comment (thehenrytsai) [reply tests/core/Blockchain.spec.ts]: Took your suggestion, checking for specific error code.

File: tests/core/Blockchain.spec.ts (hunk 3 above)
Comment (thehenrytsai) [reply tests/core/Blockchain.spec.ts]: Took your suggestion, checking for specific error code.

**Pull request merged**